_source/_posts/2023-07-27-spring-graphql-react.adoc

```diff
@@ -0,0 +1,1857 @@
+---
+layout: blog_post
+title: "How to build a GraphQL API with Spring for GraphQL"
+author: jimena-garbarino
+by: contractor
+communities: [security,java,javascript]
+description: "A step-by-step guide for building a secured GraphQL API with Spring Boot and Auth0 authentication on React"
+tags: [java, javascript]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness
+---
+:page-liquid:
+:toc: macro
+:experimental:
+
+
+GraphQL is a query language for APIs and a runtime that allows the API consumer to get exactly the required information instead of the server exclusively controlling the contents of the response. While some REST API implementations require loading the references of a resource from multiple URLs, GraphQL APIs can follow references between related objects and return them in a single request. This step-by-step guide demonstrates how to build a GraphQL API with Spring Boot and Spring for GraphQL, for querying a sample dataset of related companies, persons, and properties, seeded to a Neo4j database. It also demonstrates how to build a React client with Next.js and MUI Datagrid to consume the API. Both the client and the server are secured with Auth0 authentication using Okta Spring Boot Starter for the server and Auth0 React SDK for the client.
```

Comment:
I'd recommend adding a paragraph break before "This step-by-step..."

Reply:
Added